src/shared/components/horizontal-scroller/index.js

```diff
@@ -0,0 +1,108 @@
+import React, { PureComponent } from 'react'
+import PropTypes from 'prop-types'
+import ReactResizeDetector from 'react-resize-detector'
+import classNames from 'classnames'
+
+import ArrowButton from './arrows'
+import styles from './index.module.css'
+
+class HorizontalScroller extends PureComponent {
+  state = {
+    showSliderButtons: false,
+    isLeftArrowActive: false,
+    isRightArrowActive: true
+  }
+
+  componentDidMount () {
+    this.currentScrollX = 0
+  }
+
+  render () {
+    const { renderNavPills, className } = this.props
+    const { showSliderButtons, isLeftArrowActive, isRightArrowActive } = this.state
+    const containerClasses = classNames(styles.container, className, {
+      [styles.showButtons]: showSliderButtons
+    })
+
+    return (
+      <div className={ containerClasses }>
+        <ArrowButton className={ styles.leftButton }
+          direction="left"
+          handleOnClick={ this.handleLeftButtonClick }
+          active={ isLeftArrowActive }/>
+        <ArrowButton className={ styles.rightButton }
+          direction="right"
+          handleOnClick={ this.handleRightButtonClick }
+          active={ isRightArrowActive }/>
+        <div className={ styles.scroller } ref={ this.handleScrollerRef } onScroll={ this.handleOnScroll }>
+          <ReactResizeDetector handleWidth
+            handleHeight
+            onResize={ this.handleOnResizeScroller } />
+          <div className={ styles.pillsWrapper } ref={ this.handlePillsWrapperRef }>
+            { renderNavPills() }
```

Comment:
Why receiving a function to render the `navPills` instead of having that logic here, and a pills array prop?